explain-joins.md

```diff
@@ -0,0 +1,267 @@
+| │ └─Selection_33             | 71.01    | 10000   | cop[tikv] |               | proc max:96ms, min:40ms, p80:96ms, p95:96ms, iters:83, tasks:3                                                                                                                                                                                                          | eq(test.t1.int_col, 1)                            | N/A                   | N/A     |
+| │   └─TableFullScan_32       | 71010.00 | 71010   | cop[tikv] | table:t1      | proc max:96ms, min:40ms, p80:96ms, p95:96ms, iters:83, tasks:3                                                                                                                                                                                                          | keep order:false                                  | N/A                   | N/A     |
+| └─TableReader_36(Probe)      | 90000.00 | 90000   | root      |               | time:443.457881ms, loops:90, cop_task: {num: 3, max: 443.829268ms, min: 312.204258ms, avg: 372.765158ms, p95: 443.829268ms, max_proc_keys: 31712, p95_proc_keys: 31712, tot_proc: 580ms, tot_wait: 4ms, rpc_num: 3, rpc_time: 1.118223204s, copr_cache_hit_ratio: 0.00} | data:TableFullScan_35                             | 182.62188053131104 MB | N/A     |
+|   └─TableFullScan_35         | 90000.00 | 90000   | cop[tikv] | table:t2      | proc max:112ms, min:92ms, p80:112ms, p95:112ms, iters:102, tasks:3                                                                                                                                                                                                      | keep order:false                                  | N/A                   | N/A     |
++------------------------------+----------+---------+-----------+---------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------+-----------------------+---------+
+6 rows in set (0.37 sec)
+```
+
+If the index on `t1.int_col` is not available, the hash join execution plan remains the same. However, the plan changes for the Index Join and performance is reduced:
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE t2 DROP INDEX t1_id; # drop the index
+EXPLAIN ANALYZE SELECT /*+ INL_JOIN(t1, t2) */  * FROM t1 INNER JOIN t2 ON t1.id = t2.t1_id WHERE t1.int_col = 1;
+EXPLAIN ANALYZE SELECT /*+ HASH_JOIN(t1, t2) */  * FROM t1 INNER JOIN t2 ON t1.id = t2.t1_id WHERE t1.int_col = 1;
+ALTER TABLE t2 ADD INDEX (t1_id); # Re-add the index
+```
```

Comment:
Could we provide such an example? So this can guide users to add an index to speed up their Joins.
```sql
--- a table without index `t1_id`
EXPLAIN ANALYZE SELECT * FROM t1 INNER JOIN t2 ON t1.id = t2.t1_id WHERE t1.int_col = 1; --- run Join without hint
--- and get a slow Hash Join result
ALTER TABLE t1 ADD INDEX  t1_id (id);
EXPLAIN ANALYZE SELECT * FROM t1 INNER JOIN t2 ON t1.id = t2.t1_id WHERE t1.int_col = 1; --- run Join without hint
--- and get a fast Index Join result
```

Reply:
See https://github.com/pingcap/docs/pull/3922/commits/d3df1326fc80c128bb78ce302119467d04e7ff0a